web/modules/custom/ai_screening_project_track/src/Entity/ProjectTrack.php

```diff
@@ -179,8 +179,21 @@ public function getDescription(): string {
   /**
    * {@inheritdoc}
    */
-  public function setProjectTrackEvaluation(Evaluation $evaluation): self {
-    $this->set('project_track_evaluation', $evaluation->value);
+  public function setProjectTrackEvaluation(array $evaluations): self {
+    // Set track evaluation to the least possible approval found across the
+    // tools.
+    if (in_array(Evaluation::REFUSED, $evaluations)) {
+      $this->set('project_track_evaluation', Evaluation::REFUSED->value);
+    }
+    elseif (in_array(Evaluation::UNDECIDED, $evaluations)) {
+      $this->set('project_track_evaluation', Evaluation::UNDECIDED->value);
+    }
+    elseif (in_array(Evaluation::APPROVED, $evaluations)) {
+      $this->set('project_track_evaluation', Evaluation::APPROVED->value);
+    }
+    else {
+      $this->set('project_track_evaluation', Evaluation::UNDECIDED->value);
+    }
```

Comment:
Using `match` (https://www.php.net/manual/en/control-structures.match.php) may make this a little easier to read and write, e.g.

```suggestion
    $evaluation = match(TRUE) {
      in_array(Evaluation::REFUSED, $evaluations) => Evaluation::REFUSED->value,
      in_array(Evaluation::UNDECIDED, $evaluations) => Evaluation::UNDECIDED->value,
      in_array(Evaluation::APPROVED, $evaluations) => Evaluation::APPROVED->value,
      default => Evaluation::UNDECIDED->value,
    };
    $this->set('project_track_evaluation', $evaluation);
```